include/drivers/kinetis/i2c.h

```diff
@@ -0,0 +1,106 @@
+#pragma once
+#include <MK64F12.h>
+#include <stdbool.h>
+
```

Comment:
Overall, a little more clarity here might be good, here are some suggestions:
 - document complicated struct members
 - briefly describe the function of various macros